substrate.go

```diff
@@ -23,6 +23,15 @@ type DiscardableMessage interface {
 	DiscardPayload()
 }
 
+// AnnotatedMessage is an interface implemented by messages used by sink wrappers,
+// in particular the synchronous sink adapter. This allows the backends to retrieve
+// the original message the user sent when invoking user provided callbacks for example
+// for getting partition keys.
+type AnnotatedMessage interface {
+	Message
+	OriginalMessage() Message
```

Comment:
IMO I'd change the method name to `Original`, I think have `Message` as a suffix when it returns type `Message` is a little repetitive.